waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -913,23 +913,48 @@ proc addAll*(wakuRlnRelay: WakuRLNRelay, list: seq[IDCommitment]): RlnRelayResul
       return err(memberAdded.error())
   return ok()
 
-# the types of inputs to this handler matches the MemberRegistered event/proc defined in the MembershipContract interface
-type RegistrationEventHandler  = proc(pubkey: Uint256, index: Uint256): void {.gcsafe, closure, raises: [Defect].}
+type GroupUpdateHandler = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.gcsafe, raises: [Defect].}
 
-proc subscribeToMemberRegistrations(web3: Web3, contractAddress: Address, handler: RegistrationEventHandler, fromBlock: string = "0x0"): Future[Subscription] {.async, gcsafe} =
-  var contractObj = web3.contractSender(MembershipContract, contractAddress)
-  return await contractObj.subscribe(MemberRegistered, %*{"fromBlock": fromBlock, "address": contractAddress}) do(pubkey: Uint256, index: Uint256){.raises: [Defect], gcsafe.}:
+proc generateGroupUpdateHandler(rlnPeer: WakuRLNRelay): GroupUpdateHandler =
+  # assuming all the members arrive in order
+  # TODO: check the index and the pubkey depending on
+  # the group update operation
+  var handler: GroupUpdateHandler
+  handler = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.raises: [Defect].} =
```

Comment:
As part of the member deletion feature, this handler will soon take in an additional argument, `GroupUpdateType`, which will be either `Insertion` or `Deletion`, and then we can handle those cases separately.